File: p13/Entry4.java
interface GO{
  void Collide(GameObject go2);
}

class Ship extends GO{
  void Collide(GameObject go2){
    //this is ship
    if(go2 is instanceof(Station))
      ...logic of ship with station
    if(go2 is instanceof(Commet))
      ...logic of ship with commet
    if(go2 is instanceof(Aestroid))
      ...logic of ship with aestroid
  }
}

class Handler
{
     public void Collide(GameObject go1,GameObject go2)
     {
          go1.Collide(go2);
     }
}

File: p13/Entry4.py
class Ship:
    def collide(self, other):
        # This is Ship
        if isinstance(other, Station):
            # Logic of Ship with Station
            pass
        elif isinstance(other, Comet):
            # Logic of Ship with Comet
            pass
        elif isinstance(other, Asteroid):
            # Logic of Ship with Asteroid
            pass
class Station:
    def collide(self, other):
        # This is Station
        if isinstance(other, Ship):
            # Logic of Ship with Station
            pass
        elif isinstance(other, Comet):
            # Logic of Station with Comet
            pass
        elif isinstance(other, Asteroid):
            # Logic of Station with Asteroid
            pass

class Handler:
    def collide(self, go1, go2):
        go1.collide(go2)